fuel-vm/src/interpreter/alu/wideint.rs

```diff
@@ -0,0 +1,365 @@
+
+                    inc_pc(pc)
+                }
+
+                pub(crate) fn [<alu_wideint_mul_ $t:lower>](
+                    &mut self,
+                    dest_addr: Word,
+                    b: Word,
+                    c: Word,
+                    args: MulArgs,
+                ) -> Result<(), RuntimeError> {
+                    let owner_regs = self.ownership_registers();
+                    let (SystemRegisters { flag, mut of, mut err, pc, .. }, _) = split_registers(&mut self.registers);
+
+                    // LHS is only indirect if the flag is set
+                    let lhs: $t = if args.indirect_lhs {
+                        $t::from_be_bytes(read_bytes(&self.memory, b)?)
+                    } else {
+                        b.into()
+                    };
+                    // RHS is only indirect if the flag is set
+                    let rhs: $t = if args.indirect_rhs {
+                        $t::from_be_bytes(read_bytes(&self.memory, c)?)
+                    } else {
+                        c.into()
+                    };
+
+                    let (wrapped, overflow) = $t::overflowing_mul(lhs, rhs);
+
+                    if overflow && !is_wrapping(flag.into()) {
+                        return Err(PanicReason::ArithmeticOverflow.into());
+                    }
+
+                    *of = overflow as Word;
+                    *err = 0;
+
+                    write_bytes(&mut self.memory, owner_regs, dest_addr, wrapped.to_be_bytes())?;
+
+                    inc_pc(pc)
+                }
+
+                pub(crate) fn [<alu_wideint_div_ $t:lower>](
+                    &mut self,
+                    dest_addr: Word,
+                    b: Word,
+                    c: Word,
+                    args: DivArgs,
+                ) -> Result<(), RuntimeError> {
+                    let owner_regs = self.ownership_registers();
+                    let (SystemRegisters { flag, mut of, mut err, pc, .. }, _) = split_registers(&mut self.registers);
+
+                    // LHS is always indirect
+                    let lhs: $t = $t::from_be_bytes(read_bytes(&self.memory, b)?);
+
+                    // RHS is only indirect if the flag is set
+                    let rhs: $t = if args.indirect_rhs {
+                        $t::from_be_bytes(read_bytes(&self.memory, c)?)
+                    } else {
+                        c.into()
+                    };
+
+                    let lhs = [<to_prim_ $t:lower>](lhs);
+                    let rhs = [<to_prim_ $t:lower>](rhs);
+
+                    let result = match lhs.checked_div(rhs) {
+                        Some(d) => [<from_prim_ $t:lower>](d),
+                        None => {
+                            if is_unsafe_math(flag.into()) {
+                                $t::default() // Zero
+                            } else {
+                                return Err(PanicReason::ErrorFlag.into());
```

Comment:
`ErrorFlag` is typically used when something unexpected happens (i.e. a bug or incorrectly validated transaction). Since a user could trigger this behavior without any sort of intervention on our end, I'd go with a different kind of panic reason like `ArithmeticOverflow` or something more specific.

Reply:
This is consistent with other ALU behaviour. Would it make sense to update those as well?

Reply:
Updated.